com.ibm.streamsx.metrics/com.ibm.streamsx.metrics.service/MetricsIngestService.spl

```diff
@@ -32,38 +32,39 @@
 namespace com.ibm.streamsx.metrics.service;
 
 use com.ibm.streamsx.metrics::* ;
-use com.ibm.streamsx.json::TupleToJSON ;
+use com.ibm.streamsx.json::* ;
 use com.ibm.streamsx.topology.topic::Publish ;
 
 public composite MetricsIngestService
 {			
 	param
-		expression<rstring> $applicationConfigurationName ;
+		expression<rstring> $applicationConfigurationName : getSubmissionTimeValue("applicationConfigurationName", "metricsConfig");
```

Comment:
It actually holds the value of the JSON filters in a variable. Every time the operator produces a tuple, it gets the JSON filters from the app config and compares it to the stored one. If they don't match, the setupFilters() method is called again.